pkg/resources/kafka/kafka.go

```diff
@@ -1203,20 +1218,50 @@ func getServiceFromExternalListener(client client.Client, cluster *v1beta1.Kafka
 // - prioritize upscale in order to allow upscaling the cluster even when there is a stuck RU
 // - prioritize missing broker pods to be able for escaping from offline partitions, not all replicas in sync which
 //		could stall RU flow
-func reorderBrokers(brokerPods corev1.PodList, desiredBrokers []v1beta1.Broker, brokersState map[string]v1beta1.BrokerState, controllerBrokerID int32) []v1beta1.Broker {
+func reorderBrokers(brokerPods corev1.PodList, desiredBrokers []v1beta1.Broker, brokersState map[string]v1beta1.BrokerState, pvcList corev1.PersistentVolumeClaimList, controllerBrokerID int32, log logr.Logger) []v1beta1.Broker {
 	runningBrokers := make(map[string]struct{})
 	for _, b := range brokerPods.Items {
 		brokerID := b.GetLabels()["brokerId"]
 		runningBrokers[brokerID] = struct{}{}
 	}
 
+	presentPersistentVolumeClaims := make(map[string]struct{})
+	for _, pvc := range pvcList.Items {
```

Comment:
Checking if the broker's PVC exists alone is not enough, we need to check if the underlying PV that stores kafka log dirs exists. The status of the PVC needs to be verified to be in `Bound` state. (see `Phase` field of https://pkg.go.dev/k8s.io/api@v0.23.8/core/v1#PersistentVolumeClaimStatus)

Reply:
one remark here: in our environemnt we don't have access to the PersistentVolume resources - we can access only the PVC CR

Reply:
I don't think it's necessary to check PVs it's enough verify the status of the PVC